### DISCUSSION TITLE
run command / rule before shutterposition

### DISCUSSION THREAD

**Discussion by mczac** [arendst/Tasmota#12262] [Q&A] [answered]
Hello,
I have shutters with Tasmota and all works perfectly. I have it integrated with HomeBridge. Now what I need is to trigger a different relay on the same board just before MQTT cmnd/tasmota_xxxx/shutterposition/ is executed and relay up or down is triggered. Is it possible? Many many thanks.

**Comment by sfromis** [comment, marked as answer]
One possible avenue could be to change the commands you publish from HomeBridge to instead be `Event` commands. That allows you to create rules to react to those event commands (including a parameter), and perform any relevant sequence of commands, likely with that ShutterPosition as the last.

**Reply by mczac** [reply]
something like this?
cmnd/sonoff/event customevent parameter

I didn't check the syntax of the rule, yet. one question though. how do I send to MQTT even topic 2 parameters event name and position parameter? What's the Tasmota syntax?
Rule ON customevent DO Set Power3=ON Delay 3 set ShutterPosition = parameter ENDON

**Reply by sfromis** [reply]
Not quite. The command is [Event](https://tasmota.github.io/docs/Commands/#event), and an example rule would be:
`on Event#setpos do Backlog Power3 ON; Delay 3; ShutterPosition %value% endon`
And publishing to topic `cmnd/sonoff/Event` payload `setpos=20`

Note that while Delay will most often work out, it is not ideal for timing in rules, as it is possible for the timing to be cut short in case of something else happening, like some other command.

While using the Event command is supported "out of the box", there also exists a [Subscribe](https://tasmota.github.io/docs/Subscribe-&-Unsubscribe/) feature allowing you to also make new command names, but only exists if you build your own binary to include it.

**Reply by mczac** [reply]
Cheers!!!
